### PR TITLE
Content Type Integration

### PR DESCRIPTION
**Pull request by Snugug**
Look at all the things.

**Changelog**

```markdown
**New**

* Content Type Saving/Validation/Updating
* New Navigation

**Changed**

* Clean Up Routing
*  Clean Up Navigation
```

---
Resolves: #11 
Resolves https://github.com/punchcard-cms/punchcard/pull/40
Resolves https://github.com/punchcard-cms/punchcard/pull/36

`DCO 1.1 Signed-off-by: Sam Richard <sam@snug.ug>`

### CODE REVIEW

**Comment by Snugug** [comment]
* Probably need to write more tests
* Will need to resolve conflicts

File: content-types/services.yml [hunk 1]
```diff
@@ -1,34 +1,30 @@
 name: Services
```
Comment (scottnath) [inline content-types/services.yml]: nitpik...but in anticipation of the name change...make this `api`?

File: lib/routes/content.js [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+const multipartMiddleware = multipart();
+
+/*
+ * Content Route Resolution
+ *
+ * @param {object} application - Express Application
+ * @returns {object} - Configured Express Application
+ */
+const contentRoutes = application => {
```
Comment (scottnath) [inline lib/routes/content.js]: just `routes`

File: lib/routes/content.js [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+    });
+
+    /*
+     * Individual Content Type Landing Page
+     */
+    app.get(`/${config.content.base}/:type`, (req, res, next) => {
+      const type = utils.singleItem('id', req.params.type.toLowerCase(), types);
+
+      if (type === false) {
+        return next(new Error(`Content Type ${req.params.type} not found`));
```
Comment (scottnath) [inline lib/routes/content.js]: This error message we're going to want to be i18n friendly, so we don't want to hard-code it.

File: lib/routes/content.js [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+        delete req.session.form.content.add; // eslint-disable-line no-param-reassign
+      }
+      catch (e) {
+        req.session.form.content.add = {}; // eslint-disable-line no-param-reassign
+      }
+
+      values = utils.config(values);
+
+      if (type === false) {
+        return next(new Error(`Content Type ${req.params.type} not found`));
```
Comment (scottnath) [inline lib/routes/content.js]: put message text into config

File: lib/utils.js [hunk 5]
```diff
@@ -9,7 +12,7 @@
  *
  * @returns {object|boolean} - Will return the first filtered object, or `false` if no objects match
  */
-exports.singleItem = (key, value, arr) => {
+module.exports.singleItem = (key, value, arr) => {
```
Comment (scottnath) [inline lib/utils.js]: the `module.exports` in this file differentiates from the way other files are doing it.

**Comment by scottnath** [comment]
@Snugug I'm cool with this. Check your branch...merge issues

My issues:

1. utils is structured different from other files
2. fine to remove `foo` content type, but I suggest replacing it so we have at least two (will help with the upcoming UX happening)
3. perhaps `services` should be something more...generic?

**Comment by Snugug** [comment]
Okay @scottnath, let's file issues to update the testing content types, and I'll merge this in

**Comment by scottnath** [comment]
👍

**Comment by Snugug** [comment]
@scottnath Please take one last look at this. If you 👍 I'll merge it in

**Comment by scottnath** [comment]
@Snugug like the new 404 plan. v cool

check your branch...conflicts 

and the tests seem to be waiting for something. The next life maybe...who knows?

**Pull request closed without merging**